### PR TITLE
docs: Bump go version from 1.16 to 1.18

### PR DESCRIPTION
**Pull request by yahonda**
### What issue does this PR solve?

- close #xxxx
- to #xxxx

### What is changed:
This pull request bumps go version from 1.16 to 1.18 in the getting started page to follow up https://github.com/pingcap/tidb/issues/32874

### CODE REVIEW

File: src/get-started/install-golang.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ To build TiDB from source code, you need to install Go in your development envir
 
 ## Install Go 1.16
```
Comment (zhangyangyu) [inline src/get-started/install-golang.md:5]: one oversight

Comment (yahonda) [reply src/get-started/install-golang.md]: Fixed. Thanks for the review.

**Review by zhangyangyu — APPROVED** [review]

**Comment by zhangyangyu** [comment]
@all-contributors please add @yahonda for content

**Comment by zhangyangyu** [comment]
@all-contributors please add @yahonda for content

**Comment by yahonda** [comment]
Thanks!

**Pull request merged**